openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -414,6 +391,16 @@ def post(self, request, course_key_string, *args, **kwargs):
         """
         course_key = CourseKey.from_string(course_key_string)
 
+        # Check if we're masquerading as someone else. If so, we should just ignore this request.
```

Comment:
Does this mean a staff user has to un-masquerade before they can use this endpoint?
I thought the value persisted for some time in the `request.session`,

Reply:
I'm not sure how long it lasts exactly. But this check only prevents the endpoint from working if the staff user is specifically masquerading as someone else (not like, just as a learner or verified user or whatever).

And that's appropriate for this endpoint, since this endpoint turns off an interaction that the user should see themselves a celebration of their progress).